core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -184,6 +184,48 @@ class PlaintextConsumerTest extends BaseConsumerTest {
   }
 
   @Test
+  def testSubsequentPatternSubscription() {
+    val numRecords = 10000
+    sendRecords(numRecords)
+
+    val fooTopic = "foo" // matches both subscription patterns
+    TestUtils.createTopic(this.zkUtils, fooTopic, 1, serverCount, this.servers)
+    sendRecords(1000, new TopicPartition(fooTopic, 0))
+
+    assertEquals(0, this.consumers.head.assignment().size)
+
+    val pattern1 = Pattern.compile(".*") // both 'topic' and 'foo' match this 
+    this.consumers.head.subscribe(pattern1, new TestConsumerReassignmentListener)
+    this.consumers.head.poll(50)
+
+    var subscriptions = Set(
+      new TopicPartition(topic, 0),
+      new TopicPartition(topic, 1),
+      new TopicPartition(fooTopic, 0))
+
+    TestUtils.waitUntilTrue(() => {
+      this.consumers.head.poll(50)
+      this.consumers.head.assignment() == subscriptions.asJava
+    }, s"Expected partitions ${subscriptions.asJava} but actually got ${this.consumers.head.assignment()}")
+
+    val pattern2 = Pattern.compile("f.*") // only 'foo' matches this
```

Comment:
@ewencp Thanks for your feedback and catching the missing test. I'll add the test and the comments as you suggested.